ygot/struct_validation_map_test.go

```diff
@@ -407,3 +407,264 @@ func TestInitContainer(t *testing.T) {
 		}
 	}
 }
+
+func TestMergeJSON(t *testing.T) {
+	tests := []struct {
+		name    string
+		inA     map[string]interface{}
+		inB     map[string]interface{}
+		want    map[string]interface{}
```

Comment:
Optional: you can represent these test cases more compactly with JSON text + unmarshal.

Reply:
Agreed, I usually like to try and keep the base input type so that it'd be possible to have errors that can't occur in JSON documents be testable too.